src/_guides/language/effective-dart/design.md

```diff
@@ -658,50 +696,121 @@ can.
+    doesn't produce output, write files, etc. A getter should generally follow
+    that.
 
-{:.good-style}
-{% prettify dart %}
-rectangle.width
-collection.isEmpty
-button.canShow
-{% endprettify %}
+    The "user-visible" part is important. It's fine for getters to modify hidden
+    state or produce out of band side effects. Getters can lazily calculate and
+    store their result, write to a cache, log stuff, etc. As long as the caller
+    doesn't *care* about the side effect, it's probably fine.
 
-{:.bad-style}
+    {:.bad-style}
+    {% prettify dart %}
+    stdout.newline; // Produces output.
+    list.clear; // Modifies object.
+    {% endprettify %}
+
+*   **The operation is *idempotent*.** "Idempotent" is an odd word that, in this
+    context, basically means that calling the operation multiple times produces
+    the same result each time, unless some state is explicitly modified between
+    those calls. (Obviously, `list.length` produces different results if you add
+    an element to the list between calls.)
+
+    "Same" here does not mean a getter must literally produce an identical
+    object on successive calls. Requiring that would force many getters to have
+    brittle caching, which negates the whole point of using a getter. It's
+    common, and perfectly fine, for a getter to return a new future or list each
+    time you call it. The important part is that the future completes to the
+    same value and the list contains the same elements.
+
+    In other words, the result value should be the same *in the aspects that the
+    caller cares about.*
+
+    {:.bad-style}
+    {% prettify dart %}
+    DateTime.now; // New result each time.
+    {% endprettify %}
+
+*   **The resulting object doesn't expose all of the original object's state.**
+    A field exposes only a piece of an object. If your operation returns a
+    result that exposes the original object's entire state, it's likely better
+    off as a [`to___()`][to] or [`as___()`][as] method.
+
+[to]: #prefer-naming-a-method-to___-if-it-copies-the-objects-state-to-a-new-object
+[as]: #prefer-naming-a-method-as___-if-it-returns-a-different-representation-backed-by-the-original-object
+
+If all of the above describe your operation, it should be a getter. It seems
+like few members would survive that gauntlet, but surprisingly many do. Many
+operations just do some computation on some state and most of those can and
+should be getters.
+
+{:.good-style}
 {% prettify dart %}
-DateTime.now;   // Returns different value each call.
-window.refresh; // Doesn't return a value.
+rectangle.area;
+collection.isEmpty;
+button.canShow;
+dataSet.minimumValue;
 {% endprettify %}
 
-Unlike other languages, in Dart we don't require getters to be particularly fast
-or have certain complexity guarantees. Calling `length` on an Iterable may be
-`O(n)`, and that's OK.
-
 
 ### DO use a setter for operations that conceptually change a property.
```

Comment:
a setter -> setters
(Or change the previous heading from plural to singular: DO use a getter for an operations that conceptually accesses a property.)

Reply:
Done.